### ISSUE TITLE
Seigan Thirdeye is too powerful and lasts too long.

### ISSUE THREAD

**Issue by ShiyoKozuki** [DerpyProjectGroup/topaz#842]
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened
- [x] I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I have specified what branch this happens on branch: EDITME

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** 
With elegy / slow1/2 you can pretty much keep up Seigan + TE perma and tank nearly anything with it. This wasn't the case in retail, and why people called it unreliable and used /nin for tanking the vast majority of the time.

https://github.com/LandSandBoat/server/blob/fa9f40932020e48dbe7ad092acd4e6933f9b528a/src/map/attack.cpp#L372

From a lot, a lot, a lot of testing with our version of Seigan + TE, changing it to this makes it feel MUCH more fair for the poor helpless monsters and retail accurate:
`if (xirand::GetRandomNumber(100) < (100 - (pastAnticipations * 20) + m_victim->getMod(Mod::THIRD_EYE_ANTICIPATE_RATE)))`

**Comment by zach2good** [comment]
The rest of the block in question gives _a lot_ more context to whats going on here:
```cpp
    else
    { // do have seigan, decay anticipations correctly (guesstimated)
        // 5-6 anticipates is a 'lucky' streak, going to assume 15% decay per proc, with a 100% base w/ Seigan
        if (xirand::GetRandomNumber(100) < (100 - (pastAnticipations * 15) + m_victim->getMod(Mod::THIRD_EYE_ANTICIPATE_RATE)))
```

I had a quick look and google around and I can't find any wiki entries, blogs, videos etc. that attempt to quantify this decay rate.

If it's allowing a style of gameplay that isn't retail or era accurate, I'm in favour of nerfing this by boosting the decay rate.

However, I don't want anyone to think that a change done in this manner makes this retail accurate. We're changing a number based on a "down the line" outcome, rather than comparing it to actual retail testing. This _cannot_ be retail accurate until it's adjusted based on actual retail testing.

**Comment by ShiyoKozuki** [comment]
Yeah, the issue is it's allowing a playstyle that was completely unheard of in retail. With 20%, it lasts around 2-3 hits the vast majority of the time and very very very rarely ever lasts longer. That's pretty much retail-like behavior.

In retail, it has some fading thing that decays over time but it was never really fully tested, so this fix is about as good as we're going to get without extensive testing.

Rukenshins blog talks about it, but he is also known for saying a lot of inaccurate things so I wouldn't use his opinion as a bible.
https://rukenshin.livejournal.com/14854.html

**Comment by TeoTwawki** [comment]
Another consideration for siegan+eye : I dunno if its ever been proven but certainly seems like it has a decay over time till it wears off. Like proc rate is much better at the time immediately after its used. Some folks on bg had theorized that it probably has decay and that each time it procs without wearing off takes a bite out of its total till it an attack takes it off of you. That seems kinda overly complicated for SE though. But with unknowns like that its _incredibly_ unlikely we'll get it accurate. My suggestion is we lower the default and make it configurable.

edit: I really need to learn to refresh pages I left open before posting replies

**Comment by claywar** [comment]
There are several abilities like this, for example: Fan Dance

**Comment by 59blargedy** [comment]
myself (and @rude-jerk) did some testing on this before i implemented changes on eden. here's what we came up with in lua for ease of communication. i also hate that it's in a utils in lua for ws and in core for auto attack but i wasnt skilled enough to combine the two at the time.
our testing was pretty simple in terms of observing the number of anticipates per each seigan/third eye application and the time between third eye application and it wearing, which resulted in observing that severe drop off between 10-20 seconds remaining. we did not study if level correction had an effect, and i cannot find my source data, but we definitely tested into the hundreds of attempts with two different testers. i can at least assert that this is better than what is there now.
        if duration >= 25 and duration <= 30 then
            chanceToKeep = 95
        elseif duration >= 20 and duration < 25 then
            chanceToKeep = 90
        elseif duration >= 15 and duration < 20 then
            chanceToKeep = 70
        elseif duration >= 10 and duration < 15 then
            chanceToKeep = 20
        elseif duration >= 5 and duration < 10 then
            chanceToKeep = 15
        else
            chanceToKeep = 10
        end

**Comment by zach2good** [comment]
+1 for this approach, as long as we table-ize it or figure out a
duration/decay formula

**Comment by WinterSolstice8** [comment]
Closed by https://github.com/LandSandBoat/server/pull/8526
